### ISSUE TITLE
Accessibility Considerations for Stacker News Media Kit & Brand Guidelines v1.2 - WIP

### ISSUE THREAD

**Issue by jason-me** [stackernews/stacker.news#1922]
# Accessibility Statement

Stacker.News Brand Guidelines should be used to create accessible content for all users by following WCAG 2.2 Guidelines related to color contrast:

1.4.1 Use of Color (Level A):
Color should not be the only means of conveying information.
[WCAG 2.2 1.4.1 Use of Color](https://www.w3.org/TR/WCAG22/#use-of-color)

1.4.3 Contrast (Minimum) (Level AA):
Normal text must have at least a 4.5:1 contrast ratio, and large text at least 3:1.
[WCAG 2.2 1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG22/#contrast-minimum)
For AAA conformance, the ratios increase to 7:1 for normal text and 4.5:1 for large text (see Understanding Contrast Minimum).
[Understanding WCAG Contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)

1.4.11 Non-text Contrast (Level AA):
UI components, icons, and graphical objects must have a contrast ratio of at least 3:1 against adjacent colors.
[WCAG 2.2 1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG22/#non-text-contrast)

## Current Accessibility Issues
Some of the colors in the current color scheme present accessibility challenges, especially in light mode, where text and UI elements lack contrast, and where certain accent colors blend too closely with the background, making icons and focus indicators difficult to see. Additionally, some colors become difficult to see in both light and dark mode when threaded replies reach their max nesting (which results in a progressively lighter or darker background). 

These issues impact both text and non-text elements, and it's essential to address them to adhere to WCAG guidelines and enhance the readability and usability of the UI across both light and dark themes. For the most part colors have been chosen wisely for their high contrast and readability, but the following design guidelines are intended to ensure that all colors meet the minimum legible threshold in their use.

**Note:** For purposes of WCAG Conformance large text means text that is at least 18pt or 14pt + Bold (`font-weight: 700`) and small text refers to anything less than that.

| # | Name         | HEX code    | RGB code           | HSL                    | Small Text Usage                                                                 | Large Text Usage                                               | Icon & Focus Usage                                                        |
|---|--------------|-------------|--------------------|------------------------|----------------------------------------------------------------------------------|----------------------------------------------------------------|---------------------------------------------------------------------------|
| ![Bitcoin-only swatch](https://m.stacker.news/75602) | Bitcoin-only | `#F6911D` | `rgb(246,145,29)` | `hsl(32,92.3%,53.9%)`  | Dark: :heavy_check_mark:  OK; Light: :x: Not OK (use as accent only) | Dark: :heavy_check_mark:  OK; Light: :x: Not OK      | Dark: :heavy_check_mark:  OK; Light: :x: Not OK               |
| ![Blackr swatch](https://m.stacker.news/75590)       | Blackr       | `#000000` | `rgb(0,0,0)`       | `hsl(0,0%,0%)`         | Dark: :x: Not OK; Light: :heavy_check_mark:  OK                     | Dark: :x: Not OK; Light: :heavy_check_mark:  OK       | Dark: :x: Not OK; Light: :heavy_check_mark:  OK             |
| ![Dark swatch](https://m.stacker.news/75591)         | Dark         | `#121214` | `rgb(18,18,20)`    | `hsl(240,5.3%,7.5%)`    | Dark: :x: Not OK (identical to base); Light: :heavy_check_mark:  OK  | Dark: :x: Not OK; Light: :heavy_check_mark:  OK       | Dark: :x: Not OK; Light: :heavy_check_mark:  OK             |
| ![Gold swatch](https://m.stacker.news/75592)         | Gold         | `#FADA5E` | `rgb(250,218,94)`  | `hsl(48,94%,67.5%)`     | Dark: :heavy_check_mark:  OK; Light: :x: Not OK                     | Dark: :heavy_check_mark:  OK; Light: :x: Not OK         | Dark: :heavy_check_mark:  OK; Light: :x: Not OK               |
| ![Gray swatch](https://m.stacker.news/75593)         | Gray         | `#969696` | `rgb(150,150,150)` | `hsl(0,0%,58.8%)`       | Dark: :heavy_check_mark:  OK; Light: :x: Not OK                     | Dark: :heavy_check_mark:  OK; Light: :x: Not OK         | Dark: :heavy_check_mark:  OK; Light: :x: Not OK               |
| ![Light swatch](https://m.stacker.news/75594)        | Light        | `#F0F0F0` | `rgb(240,240,240)` | `hsl(0,0%,94.1%)`       | Dark: :heavy_check_mark:  OK; Light: :x: Not OK                     | Dark: :heavy_check_mark:  OK; Light: :x: Not OK         | Dark: :heavy_check_mark:  OK; Light: :x: Not OK               |
| ![Nostrch swatch](https://m.stacker.news/75595)      | Nostrch      | `#8C25F4` | `rgb(140,37,244)`  | `hsl(270,90.4%,55.1%)`   | Dark: :x: Not OK (only meets large text at ~3.4:1); Light: :heavy_check_mark:  OK | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK | Dark: :x: Border recommended; Light: :heavy_check_mark:  OK |
| ![Pinkish swatch](https://m.stacker.news/75596)      | Pinkish      | `#E685B5` | `rgb(230,133,181)` | `hsl(330,66%,71.2%)`    | Dark: :heavy_check_mark:  OK; Light: :x: Not OK                     | Dark: :heavy_check_mark:  OK; Light: :x: Not OK         | Dark: :heavy_check_mark:  OK; Light: :x: Not OK               |
| ![Ruby swatch](https://m.stacker.news/75597)         | Ruby         | `#C03221` | `rgb(192,50,33)`   | `hsl(6,70.7%,44.1%)`    | Dark: :x: Not OK (use for large text only); Light: :heavy_check_mark:  OK | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK | Dark: :x: Border recommended; Light: :heavy_check_mark:  OK |
| ![Sky Hover swatch](https://m.stacker.news/75598)    | Sky Hover    | `#007CBE` | `rgb(0,124,190)`   | `hsl(201,100%,37.3%)`   | Dark: :x: Not OK; Light: :x: Not OK (use for large text)                       | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK (use with caution) |
| ![Sky Links swatch](https://m.stacker.news/75599)    | Sky Links    | `#2E99D1` | `rgb(46,153,209)`  | `hsl(201,63.9%,50%)`    | Dark: :heavy_check_mark:  OK; Light: :x: Not OK                     | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK |
| ![Sky Visited swatch](https://m.stacker.news/75600)  | Sky Visited  | `#56798E` | `rgb(86,121,142)`  | `hsl(203,24.6%,44.7%)`  | Dark: :x: Not OK; Light: :heavy_check_mark:  OK                     | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK |
| ![So Fiat swatch](https://m.stacker.news/75601)      | So Fiat      | `#5C8001` | `rgb(92,128,1)`    | `hsl(77,98.4%,25.3%)`   | Dark: :x: Not OK; Light: :heavy_check_mark:  OK                     | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK | Dark: :heavy_check_mark:  OK; Light: :heavy_check_mark:  OK |

**Dark refers to dark mode baseline (#121214) & Light refers to light mode baseline (#FCFCFF)**
